### PR TITLE
69 implementing password recovery

### PR DESCRIPTION
**Pull request by IraMladanovych02**
- Added a /password/reset/ endpoint to handle password reset requests with CAPTCHA validation.
- Integrated rate limiting to prevent brute-force attempts.
- Configured email sending via Gmail SMTP to deliver password reset links.
- Customized email template (password_reset.html) to provide user-friendly instructions.
- Integrated ReCaptcha to prevent automated misuse of the reset endpoint.
- Ensured all tokens generated for password resets are time-sensitive and expire securely.

### CODE REVIEW

File: forum/templates/users/password_reset.html [hunk 1]
```diff
@@ -0,0 +1,20 @@
+<!DOCTYPE html>
+<html lang="en">
+<head>
+    <meta charset="UTF-8">
+    <title>Password Reset</title>
+</head>
+<body>
+    <h2>Відновлення пароля</h2>
```
Comment (BasilJaworski) [inline forum/templates/users/password_reset.html:8]: Just recomendation(If not nessesary then just skip this comment)

Use main tag for better html structure, wrap like:
<body>
    <main>
        <-- content -->
    </main>
</body>

File: forum/templates/users/password_reset.html [hunk 2]
```diff
@@ -0,0 +1,20 @@
+<!DOCTYPE html>
+<html lang="en">
+<head>
+    <meta charset="UTF-8">
```
Comment (BasilJaworski) [inline forum/templates/users/password_reset.html:4]: Just recommendation(If not necessary then just skip this comment)

Add meta tag to ensure the page is responsive on mobile devices
<meta name="viewport" content="width=device-width, initial-scale=1.0">

**Review by BasilJaworski — APPROVED** [review]
Added two comments how to improve the html. Just recommendation)

File: forum/users/email.py
Comment (mehalyna) [inline forum/users/email.py:1]: Currently, you are determining the protocol (http or https) based on the request's security. While this is correct, there can be scenarios where the request may not properly reflect the actual protocol, particularly behind reverse proxies or load balancers.
Use `SECURE_SSL_REDIRECT` from settings, it is better to explicitly set the protocol in your configuration.
```
from django.conf import settings

class PasswordResetEmail(email.PasswordResetEmail):
    template_name = "email/password_reset.html"

    def get_context_data(self):
        context = super().get_context_data()
        context["domain"] = get_current_site(self.context["request"]).domain
        context["protocol"] = "https" if settings.SECURE_SSL_REDIRECT else "http"
        return context
```

File: forum/users/email.py [hunk 3]
```diff
@@ -0,0 +1,12 @@
+from djoser import email
+from django.contrib.sites.shortcuts import get_current_site
+
+
+class PasswordResetEmail(email.PasswordResetEmail):
+    template_name = "email/password_reset.html"
+
+    def get_context_data(self):
+        context = super().get_context_data()
+        context["domain"] = get_current_site(self.context["request"]).domain
```
Comment (mehalyna) [inline forum/users/email.py:10]: Using `get_current_site` works well for dynamic domain detection. However, in production, it is often cleaner and more secure to explicitly define the domain in environment variables.
In `settings.py`:
```
DOMAIN_NAME = os.getenv("DOMAIN_NAME", "localhost")
```

**Review by mehalyna — CHANGES_REQUESTED** [review]

**Review by djmelyk-debelyk — APPROVED** [review]

**Review by mehalyna — APPROVED** [review]

**Pull request merged**